pr_testing/test-cmssw-provides.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash -xe
+if [ $# -ne 6 ]; then
+  echo Usage: test-cmssw-provides.sh scram_arch pkgtools_branch cmsdist_tag build_dir week_num cmssw_release
+  exit 1
+fi
+
+SCRAM_ARCH=$1
+PKG_TOOL_BRANCH=$2
+CMSDIST_TAG=$3
+BUILD_DIR=$4
+WEEK_NUM=$5
+CMSSW_RELEASE=$6
+
+SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"  # Absolute path to script
+CMS_BOT_DIR=$(dirname ${SCRIPTPATH})  # To get CMS_BOT dir path
+
+set +x
+eval $(scram unsetenv -sh)
+set -x
+
+rm -rf $WORKSPACE/test-provides
+mkdir -p $WORKSPACE/test-provides
+cd $WORKSPACE/test-provides
+if [ ! -d $WORKSPACE/pkgtools ]; then
+  git clone git@github.com:cms-sw/pkgtools --depth 1 -b $PKG_TOOL_BRANCH
+else
+  ln -s $WORKSPACE/pkgtools .
+fi
+
+if [ ! -d $WORKSPACE/cmsdist ]; then
+  git clone git@github.com:cms-sw/cmsdist --depth 1 -b $CMSDIST_TAG
+else
+  ln -s $WORKSPACE/cmsdist .
+fi
+
+# Check if we are testing with patch release
+pushd ${WORKSPACE}/${CMSSW_RELEASE}/src
+set +x
+eval $(scram runtime -sh)
+set -x
+FULL_RELEASE_BASE=$(echo ${CMSSW_FULL_RELEASE_BASE} | rev | cut -d '/' -f 1 | rev)
+if [ ${FULL_RELEASE_BASE} = ${CMSSW_RELEASE} ]; then
+  CMS_PKG_NAME='cmssw'
+else
+  CMS_PKG_NAME='cmssw-patch'
+fi
```

Comment:
note that if we are using a patch release then we need to get provides of both `cmssw and cmssw-patch` otheriwse just `cmssw` is enough

Reply:
and again this should all be done when we copied local RPM DB ( if we had copied RPM DB from /cvmfs then that should already have all the needed provides)

Reply:
I misread your initial comment as "either cmssw or cmssw-patch", not as both, sorry!

Reply:
I think duplicate entries should not be a issue. cmssw and cmssw-patch packages already has duplicates. But if rpm complains then we can `sort|uniq` the entries